2024 results (#134)
geeksblabla/stateofdev.ma: results/2024/sections/work.mdx

---
title: Work
position: 4
---

# Work

> The Moroccan IT industry continues to grow despite global economic challenges. The demand for developers remains strong, and most developers have the opportunity to find a job shortly after graduation.

Reviewer: ```suggestion
Many places and sectors in the world are facing economincal challenges, and the IT industry is no exception to that.
In Morocco however, developers are still a highly demanded asset, and most freshly-graduated easily find opportunities shortly after graduation, if not before.
```

> Remote work continues to be popular among Moroccan developers, with the majority preferring either full-time or part-time remote work options. Companies are increasingly adapting to provide more flexible work arrangements for their employees.

Reviewer: ```suggestion
Although we can notice a "Back to office" trend in big companies, which tends to persuade smaller companies into adopting similar decisions, remote work continues to be very popular among Moroccan developers, be it full-time remote or hybrid.
```

## Employment status

> The majority of respondents are employed, either full-time, part-time, or as freelancers.

Reviewer: ```suggestion
A reasonable portion of respondents are still students or looking for work, while most of them are already employed, either in companies via part-time, full-time or internship contracts, freelancers, or are engaging in entrepreneurial activities.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="work-q-0" year={2024} title={false} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="work-q-0" year={2023} title={false} />
  </TabItem>
  <TabItem title="2022">
    <Chart id="work-q-0" year={2022} title={false} />
  </TabItem>
  <TabItem title="2021">
    <Chart id="work-q-0" year={2021} title={false} />
  </TabItem>
  <TabItem title="2020">
    <Chart id="work-q-0" year={2020} title={false} />
  </TabItem>
</Tabs>

## Working overtime

> Over 24% of participants work 1-2 days of overtime per week. Is this driven by passion, dedication, or underestimated schedules?

Reviewer: ```suggestion
If some respondents never work overtime, many of them find themselves doing it at different frequencies. Reasons may be tight deadlines, passion or dedication, and even sometimes company culture.
```

<Chart id="work-q-1" title={false} />

## Job satisfaction

> Almost the same as last year with a 3% increase. Around 80% of respondents reported they are somehow happy with their current job. This is consistent with Stack Overflow's own data on developers feeling happy with their jobs.

Reviewer: ```suggestion
Most respondents are overall satisfied with their current job, even though this satisfaction has slightly decreased compared to last year's results.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="work-q-2" year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="work-q-2" year={2023} />
  </TabItem>
  <TabItem title="2022">
    <Chart id="work-q-2" year={2022} />
  </TabItem>
  <TabItem title="2021">
    <Chart id="work-q-2" year={2021} />
  </TabItem>
  <TabItem title="2020">
    <Chart id="work-q-2" year={2020} />
  </TabItem>
</Tabs>

## Side projects

> Nearly 70% of respondents engage in side projects to enhance their skills, knowledge, earn money, and invest in themselves.

Reviewer: ```suggestion
As many respondents stated that they like coding as a hobby, we find a similar proportion (70%) which is working on side projects. The motivation can be money and profit, or simply curiosity and the need to learn.
```

<Chart id="work-q-3" title={false} />

## Unemployment after graduation

> Over 60% of respondents found a job immediately after graduating, indicating that the IT sector in Morocco remains unaffected by global layoffs.

Reviewer: ```suggestion
Most respondents easily found a job right after graduation. While most companies have trouble hiring on a global scale, it seems that companies in Morocco are doing quite fine on that front.
```

  <TabItem title="3 - 4">
    <Chart
      id="work-q-8"
      title={false}
      condition={[
        {
          question_id: "profile-q-5",
          values: ["3"]
        },
        {
          question_id: "work-q-0",
          values: ["2"]
        }
      ]}
    />
  </TabItem>
  <TabItem title="5 - 6 ">
    <Chart
      id="work-q-8"
      title={false}
      condition={[
        {
          question_id: "profile-q-5",
          values: ["4"]
        },
        {
          question_id: "work-q-0",
          values: ["2"]
        }
      ]}
    />
  </TabItem>
  <TabItem title="7 - 10 ">
    <Chart
      id="work-q-8"
      title={false}
      condition={[
        {
          question_id: "profile-q-5",
          values: ["5"]
        },
        {
          question_id: "work-q-0",
          values: ["2"]
        }
      ]}
    />
  </TabItem>
  <TabItem title="10+ ">
    <Chart
      id="work-q-8"
      title={false}
      condition={[
        {
          question_id: "profile-q-5",
          values: ["6", "7"]
        },
        {
          question_id: "work-q-0",
          values: ["2"]
        }
      ]}
    />
  </TabItem>
</Tabs>

## Choosing a job offer

> Almost similar to last year, salary, technical stack, and remote work options are the top three motivators for Moroccan developers to consider a job offer. Unsurprisingly, remote work options rank third, ahead of company culture. Data shows that industry and work impact have the least weight in career decisions.

Reviewer: ```suggestion
The results are quite comparable to previous editions. Yet it seems that respondents don't consider trainings provided by the companies as essential.
Also, the industry as well as the impact of one's work is not an important parameter in choosing a job offer.
```

<Tabs>
  <TabItem title="2024">
    <Chart id="work-q-9" year={2024} />
  </TabItem>
  <TabItem title="2023">
    <Chart id="work-q-9" year={2023} />
  </TabItem>
  <TabItem title="2022">
    <Chart id="work-q-9" year={2022} />
  </TabItem>
  <TabItem title="2021">
    <Chart id="work-q-9" year={2021} />
  </TabItem>

</Tabs>

## Preferred Company size

> Almost a split between medium and big companies; only 10% prefer companies with fewer than 10 employees.

Reviewer: ```suggestion
Most respondents tend to prefer smaller sized-companies. These companies naturally favour flexbility and are less rigid in processes compared to bigger ones. Smaller companies are perhaps also much less impacted by tech debt and legacy code.
```

<Chart id="work-q-10" title={false} />

## Agile software development methodology

> More than 84% of respondents are using an agile software methodology and more than 64% of them are using scrum.

Reviewer: ```suggestion
Most respondents follow some Agile methodology in their work, with the dominant one being Scrum.
```

<Chart id="work-q-11" title={false} />

## Remote work

> More companies are adopting remote work as almost 22% provide a full remote option and more than 55% of them provide a hybrid option to their employees.

Reviewer: ```suggestion
While hybrid mode is the dominant mode currently, we notice a slight decrease of full remote positions post-covid era.
```
